### PR TITLE
Improve some GAP related code

### PR DESCRIPTION
**Pull request by fingolfin**
- Use conjugacy_classes in GaloisGrp code
- Improve some conversions
- Improve type stability for a bunch of GAP functions

### CODE REVIEW

**Comment by fingolfin** [comment]
@ThomasBreuer this is a subset of what I did last night. But a bigger change will deal with completely changing how we use `@gapwrap` (and `@gapattribute`, which I am not sure we should keep using it at all; we'll see)

File: experimental/GaloisGrp/Group.jl [hunk 1]
```diff
@@ -145,12 +145,11 @@ Determines representatives `g` for all right-cosets of `G` modulo `H`
...
-  C = GAP.Globals.ConjugacyClasses(H.X)
+  C = conjugacy_classes(H)
   cs = GAP.Globals.CycleStructurePerm(s.X)
   can = []
-  for i=1:length(C)
-    c = C[i]
-    r = GAP.Globals.Representative(c)
+  for c in C
+    r = representative(c)
     if cs == GAP.Globals.CycleStructurePerm(r)
```
Comment (ThomasBreuer) [inline experimental/GaloisGrp/Group.jl]: Now `C` is an array of Oscar classes, `c` is such a class, `r` is an Oscar permutation, thus we should write `GAP.Globals.CycleStructurePerm(r.X)`.

**Review by ThomasBreuer — COMMENTED** [review]
just one comment for the moment

File: experimental/GaloisGrp/Group.jl (hunk 1 above)
Comment (fingolfin) [reply experimental/GaloisGrp/Group.jl]: Indeed. I rewrote the rest of the function now, too

**Pull request merged**